### PR TITLE
Modernize

### PR DESCRIPTION
**Pull request by bluetech**
This PR modernizes things I could see, please see the commits.

### CODE REVIEW

File: .github/workflows/run-checks.yml [hunk 1]
```diff
@@ -18,12 +18,16 @@ jobs:
     strategy:
       matrix:
         include:
-          - environment: "py36"
-            python: "3.6"
-          - environment: "py37"
-            python: "3.7"
           - environment: "py38"
             python: "3.8"
+          - environment: "py39"
```
Comment (nicoddemus) [inline .github/workflows/run-checks.yml:23]: No need to change, but realized that `matrix.environment` is not used at all, so this matrix could be simplified:

```yaml
matrix:
  python: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```

File: pytest-{{cookiecutter.plugin_name}}/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+[build-system]
+requires = [
+  "setuptools>=61.0.0",
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "pytest-{{cookiecutter.plugin_name}}"
```
Comment (nicoddemus) [inline pytest-{{cookiecutter.plugin_name}}/pyproject.toml:8]: We do not need any directive for setuptools to work with the `src` layout? Neat.

**Review by nicoddemus — APPROVED** [review]
Great work!

File: pytest-{{cookiecutter.plugin_name}}/pyproject.toml (hunk 2 above)
Comment (bluetech) [reply pytest-{{cookiecutter.plugin_name}}/pyproject.toml:8]: Yes it has "automatic discovery" https://setuptools.pypa.io/en/latest/userguide/package_discovery.html#automatic-discovery

I verified that the sdist and wheel look good (possible I missed something but I don't think so)

**Pull request merged**